29.cpp

```diff
@@ -0,0 +1,30 @@
+#include <climits>
+#include <cmath>
+#include <iostream>
+class Solution {
+ public:
+  int divide(int dividend, int divisor) {
+    bool is_negative = false;
+    if ((dividend > 0 && divisor < 0) || (dividend < 0 && divisor > 0))
+      is_negative = true;
+    long long quotient = 0;       // NOLINT
+    long long positive_dividend;  // NOLINT
+    positive_dividend = std::abs(dividend);
+    divisor = std::abs(divisor);
```

Comment:
I think `divident = std::abs(...); divisor = std::abs(...)` is enough. Or add a comment to clarify the positiveness of divident and divisor after abs.